web/src/compose_recipient.ts

```diff
@@ -36,7 +36,10 @@ type DirectMessagesOption = {
 function composing_to_current_topic_narrow(): boolean {
     return (
         util.lower_same(compose_state.stream_name(), narrow_state.stream_name() ?? "") &&
-        util.lower_same(compose_state.topic(), narrow_state.topic() ?? "")
+        util.lower_same(
+            util.get_final_topic_display_name(compose_state.topic()),
+            util.get_final_topic_display_name(narrow_state.topic() ?? ""),
+        )
```

Comment:
I think this is the wrong algorithm for the future corner cases of having a "general chat" normal topic (not relevant for the next year). We should be comparing the pre-final/original topics, with `""` equaling `""` wherever possible; the display topic is intended to be only for display purposes.

Reply:
Agreed. Restored to the original, as that was an unnecessary change.